edge/pkg/devicetwin/dttype/types_test.go

```diff
@@ -0,0 +1,623 @@
+		t.Run(test.name, func(t *testing.T) {
+			tv := TwinVersion{
+				EdgeVersion: test.edgeVersion,
+			}
+			tv.UpdateEdgeVersion()
+			if tv.EdgeVersion != test.edgeVersion+1 {
+				t.Errorf("UpdateEdgeVersion() failed, Got= %v Want = %v", tv.EdgeVersion, test.edgeVersion+1)
+			}
+		})
+	}
+}
+
+// TestCompareWithCloud is function to test CompareWithCloud().
+func TestCompareWithCloud(t *testing.T) {
+	tests := []struct {
+		name         string
+		cloudVersion int64
+		edgeVersion  int64
+		tvCloud      TwinVersion
+		want         bool
+	}{
+		{
+			// Failure Case
+			name:         "CompareWithCloudTest-CloudNotUpdated",
+			cloudVersion: 10,
+			edgeVersion:  10,
+			tvCloud: TwinVersion{
+				CloudVersion: 11,
+				EdgeVersion:  9,
+			},
+			want: false,
+		},
+		{
+			// Success Case
+			name:         "CompareWithCloudTest-CloudUpdated",
+			cloudVersion: 10,
+			edgeVersion:  10,
+			tvCloud: TwinVersion{
+				CloudVersion: 11,
+				EdgeVersion:  11,
+			},
+			want: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tv := TwinVersion{
+				CloudVersion: test.cloudVersion,
+				EdgeVersion:  test.edgeVersion,
+			}
+			if got := tv.CompareWithCloud(test.tvCloud); got != test.want {
+				t.Errorf("TwinVersion.CompareWithCloud() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+//Test_UpdateCloudVersion tests UpdateCloudVersion().
+func Test_UpdateCloudVersion(t *testing.T) {
+	twinVersion := TwinVersion{CloudVersion: 1}
+	bytesTwinVersion, err := json.Marshal(twinVersion)
+	if err != nil {
+		t.Errorf("Error in Marshalling: %v", err)
+	}
+	expectedTwinVersion := TwinVersion{CloudVersion: 2}
+	bytesExpectedTwinVersion, err := json.Marshal(expectedTwinVersion)
+	if err != nil {
+		t.Errorf("Error in Marshalling: %v", err)
+	}
+	tests := []struct {
+		name    string
+		version string
+		want    string
+		wantErr error
+	}{
+		{
+			// Failure Case - wrong input for unmarshal
+			name:    "UpdateCloudTest-WrongInput",
+			version: "",
+			want:    "",
+			wantErr: errors.New("unexpected end of JSON input"),
+		},
+		{
+			//Success Case - correct input for unmarshal
+			name:    "UpdateCloudTest-CorrectInput",
+			version: string(bytesTwinVersion),
+			want:    string(bytesExpectedTwinVersion),
+			wantErr: nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := UpdateCloudVersion(test.version)
+			if err != nil {
+				if !reflect.DeepEqual(err.Error(), test.wantErr.Error()) {
+					t.Errorf("Error Got = %v,Want =%v", err.Error(), test.wantErr.Error())
+					return
+				}
+			} else {
+				if !reflect.DeepEqual(err, test.wantErr) {
+					t.Errorf("Error Got = %v,Want =%v", err, test.wantErr)
+					return
+				}
+			}
+			if got != test.want {
+				t.Errorf("UpdateCloudVersion() = %v, want %v", got, test.want)
+			}
+		})
+	}
+}
+
+// Test_UpdateEdgeVersion test UpdateEdgeVersion().
+func Test_UpdateEdgeVersion(t *testing.T) {
```

Comment:
see preivous comment , remove underscore.

Reply:
I think this underscore is there because we have 2 functions with same name. Can you suggest naming conventions to follow in such cases ?
``` 
// function 1
func UpdateCloudVersion(version string) (string, error) {

// function 2
func (tv TwinVersion) UpdateCloudVersion() {
```